### PR TITLE
Add: generate servers.description & servers.variables

### PR DESCRIPTION
**Pull request by sun-yryr**
This PR enables the generation of server configurations that conform to the openapi specification.
[Here](https://swagger.io/specification/#server-object) is the reference I used.

Example:

```
config/openapi.php
~~~
'servers' => [
    [
        'url' => env('APP_URL'),
        'description' => null,
        'variables' => [
            'variable_name_A' => [
                'default' => 'default_A',
                'description' => 'sample',
                'enum' => ['A', 'B'],
            ],
            'variable_name_B' => [
                'default' => 'default_B',
                'description' => 'sample',
            ],
        ],
    ],
],
~~~
```

output json

```
~~~
"servers": [
    {
        "url": "http://localhost",
        "description": "",
        "variables": {
            "variable_name_A": {
                "description": "sample",
                "default": "default_A",
                "enum": [
                    "A",
                    "B"
                ]
            },
            "variable_name_B": {
                "description": "sample",
                "default": "default_B"
            }
        }
    }
],
~~~
```

### CODE REVIEW

File: src/Builders/ServersBuilder.php [hunk 1]
```diff
@@ -13,7 +15,26 @@ class ServersBuilder
     public function build(array $config): array
     {
         return collect($config)
-            ->map(static fn(array $server) => Server::create()->url($server['url']))
+            ->map(static function (array $server) {
+                $variables = collect(Arr::get($server, 'variables'))
+                    ->map(function (array $variable, string $key) {
+                        $tmp = ServerVariable::create($key)
```
Comment (borisson) [inline src/Builders/ServersBuilder.php:21]: `$tmp` is not a great variable name, `$serverVariable` is probably better.

File: src/Builders/ServersBuilder.php [hunk 2]
```diff
@@ -13,7 +15,26 @@ class ServersBuilder
...
+                            ->default(Arr::get($variable, 'default'))
+                            ->description(Arr::get($variable, 'description'));
+                        if (is_array(Arr::get($variable, 'enum'))) {
+                            // exist
```
Comment (borisson) [inline src/Builders/ServersBuilder.php:25]: This comment does not add anything, it should probably be removed.

File: tests/Builders/ServersBuilderTest.php [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+     * @param array $config
+     * @param array $expected
+     * @return void
+     */
+    public function testBuild(array $config, array $expected): void
+    {
+        $SUT = new ServersBuilder();
+        $servers = $SUT->build($config);
+        $this->assertSameAssociativeArray($expected[0], $servers[0]->toArray());
+        var_dump($servers[0]->toArray());
```
Comment (borisson) [inline tests/Builders/ServersBuilderTest.php:23]: Is this `var_dump` a remainder of testing/development? Should probably not end up here.

**Review by borisson — COMMENTED** [review]
Not sure if it is appropriate for me to review this, but I found some things that could be improved.

File: src/Builders/ServersBuilder.php (hunk 1 above)
Comment (vyuldashev) [reply src/Builders/ServersBuilder.php:21]: Agree

File: src/Builders/ServersBuilder.php (hunk 2 above)
Comment (vyuldashev) [reply src/Builders/ServersBuilder.php:25]: + `if` can be refactored to `->when(is_array...`

File: tests/Builders/ServersBuilderTest.php [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+{
+    /**
+     * @dataProvider providerBuild
+     * @param array $config
+     * @param array $expected
+     * @return void
+     */
+    public function testBuild(array $config, array $expected): void
+    {
+        $SUT = new ServersBuilder();
```
Comment (vyuldashev) [inline tests/Builders/ServersBuilderTest.php:20]: what does `$SUT` mean?

Comment (borisson) [reply tests/Builders/ServersBuilderTest.php:20]: [System under test](https://en.wikipedia.org/wiki/System_under_test), it is the thing being tested.
I use this personally often as well, but mostly when building several mocked objects in a test, to make it more clear which of the objects you've created is the one actually being tested. 
This is not really needed in this test.

File: tests/Builders/ServersBuilderTest.php (hunk 3 above)
Comment (sun-yryr) [reply tests/Builders/ServersBuilderTest.php:23]: Thank you. I forgot.

File: src/Builders/ServersBuilder.php (hunk 2 above)
Comment (sun-yryr) [reply src/Builders/ServersBuilder.php]: I don't know how to refactor using `->when(...` , if you don't mind.

**Comment by vyuldashev** [comment]
Released under v1.3.0. Thanks for the contribution!

**Pull request merged**